File: 0-easy/lettercase_percentage_ratio/lettercase_percentage_ratio.rb
File.open(ARGV[0]).each_line do |line|
  lowercase = 0
  uppercase = 0
  word = line.strip
  word.chars.each do |char|
    if /[[:upper:]]/.match(char)
      uppercase += 1
    else
      lowercase += 1
    end
  end
  lowercase_percent = "%.2f" % (lowercase.to_f / word.size * 100)
  uppercase_percent = "%.2f" % (uppercase.to_f / word.size * 100)
  puts "lowercase: #{lowercase_percent} uppercase: #{uppercase_percent}"
end

File: 0-easy/lettercase_percentage_ratio/lettercase_percentage_ratio.go
package main

import "fmt"
import "log"
import "bufio"
import "os"

func main() {
    file, err := os.Open(os.Args[1])
    if err != nil {
        log.Fatal(err)
    }   
    defer file.Close()
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        str := scanner.Text()
        lowercase, uppercase := 0,0
        
        for _, code := range str {
            if code >= 97 {
                lowercase += 1
            } else {
                uppercase += 1
            }
        }
        lowercasePercent := float64(lowercase) / float64(len(str)) * 100
        uppercasePercent := float64(uppercase) / float64(len(str)) * 100
        fmt.Printf("lowercase: %.2f uppercase: %.2f\n", lowercasePercent, uppercasePercent)
    }   
}